### PR TITLE
Hack path to GCC in fortran environment

### PR DESCRIPTION
**Pull request by lxavier**
This is required for serialbox

### CODE REVIEW

File: env.arolla.sh [hunk 1]
```diff
@@ -347,9 +357,14 @@ EOF
     export OLD_LD_LIBRARY_PATH=$LD_LIBRARY_PATH
     export LD_LIBRARY_PATH=${CRAY_LD_LIBRARY_PATH}:${LD_LIBRARY_PATH}
 
-    # We have gcc for gnu, cray and pgi environments
-    export CXX=g++
-    export CC=gcc
+    # Always use gcc for C and C++ compilation within Fortran environment
+    # this is required by serialbox
+    if [ -n "${GCC_PATH}" ]; then
```
Comment (havogt) [inline env.arolla.sh]:
```suggestion
    if [ -z "${GCC_PATH}" ]; then
```

**Comment by lxavier** [comment]
do we need to fix the other machine too (or is kesch ok ?)

**Comment by havogt** [comment]
works for:
- tsa: gnu, pgi
- arolla: gnu

all other machines also don't work

**Comment by lxavier** [comment]
ok so I'll do the same for kesch and daint, Still on the long run we should separate the Fortran and C++ compilation in serialbox (or is it possible already today).  So that we can load independent environment. Alternatively we need to get serialbox c++ to work with PGI anc Cray (I suppose the first option is easier)

**Comment by lxavier** [comment]
@havogt kesch is now working (I launch the plan). 
Daint is not working because of a boost issue, but the gcc path should be fine.
I will add a request for 1. in the serialbox (see my previous comment), if you think it is feasible but not yet possible with the current Cmake

**Comment by lxavier** [comment]
i removed the -b hack_C_CXX_compiler from the Jenkins plan script

**Pull request merged**